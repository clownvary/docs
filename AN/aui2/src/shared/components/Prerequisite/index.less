/* stylelint-disable */

.prerequisite {
  .prerequisite-warning {
    margin-bottom: 20px;

    &-content {
      display: inline-block;
      padding-right: 30px;
      padding-left: 10px;
      vertical-align: top;
    }

    &-override {
      padding-left: 10px;
      color: #313741;
    }
  }

  .override-field {
    margin-bottom: 20px;

    input:-webkit-autofill {
      background-color: #fff !important;
    }

    &__user-name {
      margin-bottom: 15px;
    }

    &__label {
      display: inline-block;
      width: 125px;
      padding-right: 24px;
      font-weight: 600;
      text-align: right;

      &-error {
        color: #db3811;
      }
    }

    &__value {
      display: inline-block;
      width: 235px;
      vertical-align: middle;
    }
  }
}
