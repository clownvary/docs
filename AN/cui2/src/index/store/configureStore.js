import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import { presetRedux } from 'react-base-ui/lib/middlewares';
import { redirectMiddleware, legacyCuiMiddleware } from 'shared/middlewares';
import rootReducer from 'index/reducers';

export default (initialState = {}, history) => {
  const middlewareList = [
    redirectMiddleware, ...presetRedux, legacyCuiMiddleware, routerMiddleware(history)
  ];

  const composeEnhancers =
    typeof window === 'object' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
      // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
    }) : compose;

  const middleware = __DEV__ || __STATIC__ ?
   composeEnhancers(applyMiddleware(...middlewareList)) : applyMiddleware(...middlewareList);

  const store = middleware(createStore)(rootReducer, initialState);

  if ((__DEV__ || __STATIC__) && module.hot) {
    module.hot.accept('../reducers', () => {
      store.replaceReducer(require('../reducers')); // eslint-disable-line global-require
    });
  }

  return store;
};
