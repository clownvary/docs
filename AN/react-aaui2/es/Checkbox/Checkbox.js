import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$defineProperties from 'babel-runtime/core-js/object/define-properties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import { bool, string, object, func, node, any } from 'prop-types';

import CheckboxComponent from './CheckboxComponent';
import { identity } from '../shared/utils';

var Checkbox = function (_Component) {
  _inherits(Checkbox, _Component);

  function Checkbox() {
    var _temp, _this, _ret;

    _classCallCheck(this, Checkbox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.setRef = function (checkboxComponent) {
      _this.checkboxComponent = checkboxComponent;
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  Checkbox.prototype.componentDidMount = function componentDidMount() {
    _Object$defineProperties(this, {
      checked: {
        get: function get() {
          return this.checkboxComponent.checked;
        },
        set: function set(v) {
          this.checkboxComponent.checked = !!v;
        }
      }
    });
  };

  Checkbox.prototype.render = function render() {
    var _props = this.props,
        children = _props.children,
        rest = _objectWithoutProperties(_props, ['children']);

    var auiCheckboxGroup = this.context.auiCheckboxGroup;

    // Proxy `onChange`, `checked` and `disabled` to `auiCheckboxGroup`

    if (auiCheckboxGroup) {
      rest.onChange = auiCheckboxGroup.onChange;
      rest.checked = auiCheckboxGroup.value && auiCheckboxGroup.value.indexOf(this.props.value) !== -1;
      rest.disabled = this.props.disabled || auiCheckboxGroup.disabled;
      rest.size = this.props.size || auiCheckboxGroup.size;
      rest.name = this.props.name || auiCheckboxGroup.name;
    }

    return React.createElement(
      CheckboxComponent,
      _extends({ ref: this.setRef }, rest),
      children
    );
  };

  return Checkbox;
}(Component);

Checkbox.displayName = 'AUICheckbox';
Checkbox.propTypes = {
  disabled: bool,
  size: string,
  name: string,
  children: node,
  onChange: func,
  value: any // eslint-disable-line
};
Checkbox.defaultProps = {
  onChange: identity
};
Checkbox.contextTypes = {
  auiCheckboxGroup: object };
export default Checkbox;