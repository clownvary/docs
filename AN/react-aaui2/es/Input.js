import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$defineProperties from 'babel-runtime/core-js/object/define-properties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { PureComponent } from 'react';
import { bool, string, func, object, node, oneOfType, oneOf, element } from 'prop-types';
import classNames from 'classnames';

import { identity, omit } from './shared/utils';

var Input = function (_PureComponent) {
  _inherits(Input, _PureComponent);

  function Input() {
    var _temp, _this, _ret;

    _classCallCheck(this, Input);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _PureComponent.call.apply(_PureComponent, [this].concat(args))), _this), _this.setWrappedComponentInstance = function (input) {
      _this.input = input;

      _this.props.inputRef(input);
    }, _this.handleChange = function (e) {
      var _this$props = _this.props,
          disabled = _this$props.disabled,
          onChange = _this$props.onChange;


      if (disabled) {
        return;
      }

      // If you want to access the event properties in an asynchronous way,
      // you should call `event.persist()`` on the event,
      // which will remove the synthetic event from the pool
      // and allow references to the event to be retained by user code.
      e.persist && e.persist();

      onChange(e);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  Input.prototype.componentDidMount = function componentDidMount() {
    _Object$defineProperties(this, {
      value: {
        get: function get() {
          return this.input.value;
        },
        set: function set(v) {
          if (this.props.value === undefined) {
            this.input.value = v;
          }
        }
      }
    });
  };

  // If user doesn't pass in `value` and only use `defaultValue` as init value or not, then
  // saving it to state


  Input.prototype.render = function render() {
    var _classNames;

    var _props = this.props,
        type = _props.type,
        size = _props.size,
        preIcon = _props.preIcon,
        preText = _props.preText,
        postIcon = _props.postIcon,
        postText = _props.postText,
        errored = _props.errored,
        disabled = _props.disabled,
        icon = _props.icon,
        className = _props.className,
        style = _props.style,
        PreComponent = _props.PreComponent,
        PostComponent = _props.PostComponent,
        rest = _objectWithoutProperties(_props, ['type', 'size', 'preIcon', 'preText', 'postIcon', 'postText', 'errored', 'disabled', 'icon', 'className', 'style', 'PreComponent', 'PostComponent']);

    var inputClassName = classNames((_classNames = {
      'input-group': true
    }, _classNames['input-group--' + size] = true, _classNames['input-group--icon'] = icon, _classNames['input-group--error'] = errored, _classNames['input-group--disabled'] = disabled, _classNames), className);

    return React.createElement(
      'div',
      { className: inputClassName, style: style },
      PreComponent && React.createElement(PreComponent, { className: 'input-group__item' }),
      (preIcon || preText) && React.createElement(
        'span',
        { className: 'input-group__item' },
        preIcon && React.createElement('i', { className: 'icon ' + preIcon }),
        preText
      ),
      React.createElement('input', _extends({
        // the event handler for `onChange` not firing properly in React 15.2.0 + IE11
        // when paste text into textarea: https://github.com/facebook/react/issues/7211
        // will be removed when react fixed this issue
        onInput: this.handleChange
      }, omit(rest, ['inputRef']), {
        type: type,
        disabled: disabled,
        className: 'input input-group__field',
        ref: this.setWrappedComponentInstance,
        onChange: this.handleChange
      })),
      (postIcon || postText) && React.createElement(
        'span',
        { className: 'input-group__item' },
        postIcon && React.createElement('i', { className: '' + postIcon }),
        postText
      ),
      PostComponent && React.createElement(PostComponent, { className: 'input-group__item' })
    );
  };

  return Input;
}(PureComponent);

Input.displayName = 'AUIInput';
Input.propTypes = {
  value: string,
  defaultValue: string,
  type: string.isRequired,
  size: oneOf(['sm', 'm', 'lg']).isRequired,
  preIcon: string,
  preText: string,
  postIcon: string,
  postText: string,
  errored: bool,
  icon: bool, // Represent `icon-input`
  disabled: bool,
  className: string,
  style: object, // eslint-disable-line
  onChange: func,
  PreComponent: oneOfType([func, element]),
  PostComponent: oneOfType([func, element]),
  inputRef: func,
  children: node
};
Input.defaultProps = {
  type: 'text',
  size: 'm',
  icon: false,
  onChange: identity,
  inputRef: identity
};
export default Input;