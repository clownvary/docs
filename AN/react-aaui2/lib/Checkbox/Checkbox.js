'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _defineProperties = require('babel-runtime/core-js/object/define-properties');

var _defineProperties2 = _interopRequireDefault(_defineProperties);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _CheckboxComponent = require('./CheckboxComponent');

var _CheckboxComponent2 = _interopRequireDefault(_CheckboxComponent);

var _utils = require('../shared/utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function (_Component) {
  (0, _inherits3.default)(Checkbox, _Component);

  function Checkbox() {
    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, Checkbox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.setRef = function (checkboxComponent) {
      _this.checkboxComponent = checkboxComponent;
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  Checkbox.prototype.componentDidMount = function componentDidMount() {
    (0, _defineProperties2.default)(this, {
      checked: {
        get: function get() {
          return this.checkboxComponent.checked;
        },
        set: function set(v) {
          this.checkboxComponent.checked = !!v;
        }
      }
    });
  };

  Checkbox.prototype.render = function render() {
    var _props = this.props,
        children = _props.children,
        rest = (0, _objectWithoutProperties3.default)(_props, ['children']);
    var auiCheckboxGroup = this.context.auiCheckboxGroup;

    // Proxy `onChange`, `checked` and `disabled` to `auiCheckboxGroup`

    if (auiCheckboxGroup) {
      rest.onChange = auiCheckboxGroup.onChange;
      rest.checked = auiCheckboxGroup.value && auiCheckboxGroup.value.indexOf(this.props.value) !== -1;
      rest.disabled = this.props.disabled || auiCheckboxGroup.disabled;
      rest.size = this.props.size || auiCheckboxGroup.size;
      rest.name = this.props.name || auiCheckboxGroup.name;
    }

    return _react2.default.createElement(
      _CheckboxComponent2.default,
      (0, _extends3.default)({ ref: this.setRef }, rest),
      children
    );
  };

  return Checkbox;
}(_react.Component);

Checkbox.displayName = 'AUICheckbox';
Checkbox.propTypes = {
  disabled: _propTypes.bool,
  size: _propTypes.string,
  name: _propTypes.string,
  children: _propTypes.node,
  onChange: _propTypes.func,
  value: _propTypes.any // eslint-disable-line
};
Checkbox.defaultProps = {
  onChange: _utils.identity
};
Checkbox.contextTypes = {
  auiCheckboxGroup: _propTypes.object };
exports.default = Checkbox;
module.exports = exports['default'];