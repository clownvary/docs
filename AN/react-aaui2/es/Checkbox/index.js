import Checkbox from './Checkbox';
import CheckboxGroup from './CheckboxGroup';

Checkbox.Group = CheckboxGroup;

export { Checkbox, CheckboxGroup };
export default Checkbox;