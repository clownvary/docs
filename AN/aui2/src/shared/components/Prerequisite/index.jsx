import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import Alert from 'react-base-ui/lib/components/Alert';
import Input from 'react-base-ui/lib/components/Input';
import Checkbox from 'react-base-ui/lib/components/Checkbox';
import UIComponent from 'shared/components/UIComponent';

import './index.less';

export default class Prerequisite extends UIComponent {
  /* eslint-disable react/forbid-prop-types */
  static propTypes = {
    prerequisite: PropTypes.object
  };
  /* eslint-disable react/forbid-prop-types */

  render() {
    const {
      prerequisite,
      className
    } = this.props;
    const wrapperClassName = classNames(
      'prerequisite',
      className
    );
    const haveOverrideAuthority = prerequisite.get('haveOverrideAuthority');
    const needOverride = prerequisite.get('needOverride');
    const isOverride = prerequisite.get('isOverride');
    const warningMessage = prerequisite.get('overrideMessage');

    const showPrerequisite = needOverride || (isOverride && !haveOverrideAuthority);
    return (
      <div>
        {
          showPrerequisite ? <div className={wrapperClassName}>
            { needOverride &&
              this.renderWarning(warningMessage, isOverride)
            }
            { (isOverride && !haveOverrideAuthority) &&
              this.renderUser()
            }
          </div> : null
        }
      </div>

    );
  }

  renderWarning(warningMessage, isOverride) {
    return (
      <Alert
        type="warning"
        className="prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
        noClose
      >
        <div className="prerequisite-warning-content">
          {warningMessage}
          <Checkbox
            className="prerequisite-warning-override"
            checked={isOverride}
            onChange={() => this.updateOverride(!isOverride)}
          >Override</Checkbox>
        </div>
      </Alert>
    );
  }

  renderUser() {
    let errIndex = -1;
    const { errors, userName, userPassword } = this.props.prerequisite.toJS();
    const userErr = errors.filter((err, index) => {
      /* istanbul ignore next */
      if (err.type === 'user') {
        errIndex = index;
        return true;
      }

      return false;
    });
    const isShowErr = userErr.length;

    return (
      <div className="override-field">
        <div className="override-field__user-name">
          <span className={`override-field__label ${isShowErr ? 'override-field__label-error' : ''}`}>Override User</span>
          <Input
            className="override-field__value"
            errored={isShowErr}
            autoComplete="off"
            defaultValue={userName}
            onInput={() => this.clearErr(isShowErr, errIndex)}
            onBlur={e => this.props.updateUserNameAction(e.target.value)}
          />
        </div>
        <div>
          <span className={`override-field__label ${isShowErr ? 'override-field__label-error' : ''}`}>Password</span>
          <Input
            type="password"
            className="override-field__value"
            autoComplete="off"
            defaultValue={userPassword}
            errored={isShowErr}
            onInput={() => this.clearErr(isShowErr, errIndex)}
            onBlur={e => this.props.updateUserPasswordAction(e.target.value)}
          />
        </div>
      </div>
    );
  }

  clearErr(isShowErr, errIndex) {
    /* istanbul ignore next */
    if (isShowErr) {
      this.props.clearPrerequisiteErrsAction(errIndex);
    }
  }

  updateOverride(override) {
    this.props.updateIsOverrideAction(override);
    /* istanbul ignore next */
    this.props.clearPrerequisiteErrsAction();
  }
}
