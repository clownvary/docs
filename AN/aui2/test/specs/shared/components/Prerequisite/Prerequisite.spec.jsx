import React from 'react';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import Prerequisite from 'shared/components/Prerequisite';
import Input from 'react-base-ui/lib/components/Input';
import Checkbox from 'react-base-ui/lib/components/Checkbox';

describe('shared/components/Prerequisite', () => {

  let store;

  beforeEach(() => {
    const mockStore = configureStore();
    store = mockStore();
  });

  afterEach(() => {
    store.clearActions();
  });
  const actions = {
    updateUserNameAction: jest.fn(),
    updateUserPasswordAction: jest.fn(),
    clearPrerequisiteErrsAction: jest.fn(),
    updateIsOverrideAction: jest.fn()
  }
  const setup = props => mount(
    <Prerequisite display {...props} store={store} {...actions}/>
  );

  it('Prerequisite component should render correctly if no need override', () => {
    const props = {
      prerequisite: fromJS({
        errors: [],
        needOverride: false,
        userName: '',
        userPassword: '',
        isOverride: false,
        haveOverrideAuthority: false,
        overrideMessage: ''
      })
    };
    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Prerequisite component should render correctly if need override and hasn`t checked override', () => {
    const props = {
      prerequisite: fromJS({
        errors: [],
        needOverride: true,
        userName: '',
        userPassword: '',
        isOverride: false,
        haveOverrideAuthority: false,
        overrideMessage: 'override message'
      })
    };
    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();

    component.find(Checkbox).find('input').simulate('change');
  });

  it('Prerequisite component should work fine if need override and has checked override and no authority', () => {
    const props = {
      prerequisite: fromJS({
        errors: [{
          type: 'override'
        }, {
          type: 'user'
        }],
        needOverride: true,
        userName: '',
        userPassword: '',
        isOverride: true,
        haveOverrideAuthority: false,
        overrideMessage: 'override message'
      }),
      className: 'Prerequiste--refund'
    };
    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();

    const userNameInput = component.find(Input).at(0).find('input');
    const userPasswordInput = component.find(Input).at(1).find('input');
    const overrideCheckbox = component.find(Checkbox).find('input');
    overrideCheckbox.simulate('change');
    userNameInput.simulate('blur');
    userNameInput.simulate('input');
    userPasswordInput.simulate('blur');
    userPasswordInput.simulate('input');
  });

  it('Prerequisite component should work fine if need override and has checked override and no authority and has no error', () => {
    const props = {
      prerequisite: fromJS({
        errors: [],
        needOverride: true,
        userName: '',
        userPassword: '',
        isOverride: true,
        haveOverrideAuthority: false,
        overrideMessage: 'override message'
      })
    };

    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Prerequisite component should work fine if need override and has checked overrid and has authority', () => {
    const props = {
      prerequisite: fromJS({
        errors: [],
        needOverride: true,
        userName: '',
        userPassword: '',
        isOverride: true,
        haveOverrideAuthority: true,
        overrideMessage: 'override message'
      })
    };

    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();
  });
});
