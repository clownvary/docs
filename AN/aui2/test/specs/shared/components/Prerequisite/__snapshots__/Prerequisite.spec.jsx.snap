// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared/components/Prerequisite Prerequisite component should render correctly if need override and hasn\`t checked override 1`] = `
<Prerequisite
  clearPrerequisiteErrsAction={[Function]}
  display={true}
  prerequisite={
    Immutable.Map {
      errors: Immutable.List [
      ],
      needOverride: true,
      userName: "",
      userPassword: "",
      isOverride: false,
      haveOverrideAuthority: false,
      overrideMessage: "override message",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateIsOverrideAction={[Function]}
  updateUserNameAction={[Function]}
  updateUserPasswordAction={[Function]}
>
  <div>
    <div
      className="prerequisite"
    >
      <Alert
        className="prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
        inverse={false}
        noClose={true}
        type="warning"
      >
        <div
          className="alert alert-warning prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
          role="alert"
        >
          <span
            aria-label="warning icon"
            className="icon icon-exclamation-circle"
          />
          <div
            className="prerequisite-warning-content"
          >
            override message
            <Checkbox
              checked={false}
              className="prerequisite-warning-override"
              onChange={[Function]}
            >
              <CheckboxComponent
                checked={false}
                className="prerequisite-warning-override"
                onChange={[Function]}
              >
                <label
                  className="checkbox-wrapper prerequisite-warning-override"
                  htmlFor="checkbox_attr_for_1"
                >
                  <span
                    className="checkbox"
                  >
                    <input
                      aria-checked={false}
                      checked={false}
                      className="checkbox__input"
                      id="checkbox_attr_for_1"
                      onChange={[Function]}
                      role="checkbox"
                      type="checkbox"
                    />
                    <span
                      className="checkbox__inner"
                    />
                    <span
                      className="checkbox__text"
                    >
                      Override
                    </span>
                  </span>
                </label>
              </CheckboxComponent>
            </Checkbox>
          </div>
        </div>
      </Alert>
    </div>
  </div>
</Prerequisite>
`;

exports[`shared/components/Prerequisite Prerequisite component should render correctly if no need override 1`] = `
<Prerequisite
  clearPrerequisiteErrsAction={[Function]}
  display={true}
  prerequisite={
    Immutable.Map {
      errors: Immutable.List [
      ],
      needOverride: false,
      userName: "",
      userPassword: "",
      isOverride: false,
      haveOverrideAuthority: false,
      overrideMessage: "",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateIsOverrideAction={[Function]}
  updateUserNameAction={[Function]}
  updateUserPasswordAction={[Function]}
>
  <div />
</Prerequisite>
`;

exports[`shared/components/Prerequisite Prerequisite component should work fine if need override and has checked overrid and has authority 1`] = `
<Prerequisite
  clearPrerequisiteErrsAction={[Function]}
  display={true}
  prerequisite={
    Immutable.Map {
      errors: Immutable.List [
      ],
      needOverride: true,
      userName: "",
      userPassword: "",
      isOverride: true,
      haveOverrideAuthority: true,
      overrideMessage: "override message",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateIsOverrideAction={[Function]}
  updateUserNameAction={[Function]}
  updateUserPasswordAction={[Function]}
>
  <div>
    <div
      className="prerequisite"
    >
      <Alert
        className="prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
        inverse={false}
        noClose={true}
        type="warning"
      >
        <div
          className="alert alert-warning prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
          role="alert"
        >
          <span
            aria-label="warning icon"
            className="icon icon-exclamation-circle"
          />
          <div
            className="prerequisite-warning-content"
          >
            override message
            <Checkbox
              checked={true}
              className="prerequisite-warning-override"
              onChange={[Function]}
            >
              <CheckboxComponent
                checked={true}
                className="prerequisite-warning-override"
                onChange={[Function]}
              >
                <label
                  className="checkbox-wrapper prerequisite-warning-override"
                  htmlFor="checkbox_attr_for_4"
                >
                  <span
                    className="checkbox checkbox--checked"
                  >
                    <input
                      aria-checked={true}
                      checked={true}
                      className="checkbox__input"
                      id="checkbox_attr_for_4"
                      onChange={[Function]}
                      role="checkbox"
                      type="checkbox"
                    />
                    <span
                      className="checkbox__inner"
                    />
                    <span
                      className="checkbox__text"
                    >
                      Override
                    </span>
                  </span>
                </label>
              </CheckboxComponent>
            </Checkbox>
          </div>
        </div>
      </Alert>
    </div>
  </div>
</Prerequisite>
`;

exports[`shared/components/Prerequisite Prerequisite component should work fine if need override and has checked override and no authority 1`] = `
<Prerequisite
  className="Prerequiste--refund"
  clearPrerequisiteErrsAction={[Function]}
  display={true}
  prerequisite={
    Immutable.Map {
      errors: Immutable.List [
        Immutable.Map {
          type: "override",
        },
        Immutable.Map {
          type: "user",
        },
      ],
      needOverride: true,
      userName: "",
      userPassword: "",
      isOverride: true,
      haveOverrideAuthority: false,
      overrideMessage: "override message",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateIsOverrideAction={[Function]}
  updateUserNameAction={[Function]}
  updateUserPasswordAction={[Function]}
>
  <div>
    <div
      className="prerequisite Prerequiste--refund"
    >
      <Alert
        className="prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
        inverse={false}
        noClose={true}
        type="warning"
      >
        <div
          className="alert alert-warning prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
          role="alert"
        >
          <span
            aria-label="warning icon"
            className="icon icon-exclamation-circle"
          />
          <div
            className="prerequisite-warning-content"
          >
            override message
            <Checkbox
              checked={true}
              className="prerequisite-warning-override"
              onChange={[Function]}
            >
              <CheckboxComponent
                checked={true}
                className="prerequisite-warning-override"
                onChange={[Function]}
              >
                <label
                  className="checkbox-wrapper prerequisite-warning-override"
                  htmlFor="checkbox_attr_for_2"
                >
                  <span
                    className="checkbox checkbox--checked"
                  >
                    <input
                      aria-checked={true}
                      checked={true}
                      className="checkbox__input"
                      id="checkbox_attr_for_2"
                      onChange={[Function]}
                      role="checkbox"
                      type="checkbox"
                    />
                    <span
                      className="checkbox__inner"
                    />
                    <span
                      className="checkbox__text"
                    >
                      Override
                    </span>
                  </span>
                </label>
              </CheckboxComponent>
            </Checkbox>
          </div>
        </div>
      </Alert>
      <div
        className="override-field"
      >
        <div
          className="override-field__user-name"
        >
          <span
            className="override-field__label override-field__label-error"
          >
            Override User
          </span>
          <Input
            autoComplete="off"
            className="override-field__value"
            defaultValue=""
            errored={1}
            icon={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            size="md"
            type="text"
          >
            <div
              className="input-group input-group--md input-group--error override-field__value"
            >
              <input
                autoComplete="off"
                className="input input-group__field input__field"
                defaultValue=""
                onBlur={[Function]}
                onChange={[Function]}
                onInput={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                type="text"
              />
            </div>
          </Input>
        </div>
        <div>
          <span
            className="override-field__label override-field__label-error"
          >
            Password
          </span>
          <Input
            autoComplete="off"
            className="override-field__value"
            defaultValue=""
            errored={1}
            icon={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            size="md"
            type="password"
          >
            <div
              className="input-group input-group--md input-group--error override-field__value"
            >
              <input
                autoComplete="off"
                className="input input-group__field input__field"
                defaultValue=""
                onBlur={[Function]}
                onChange={[Function]}
                onInput={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                type="password"
              />
            </div>
          </Input>
        </div>
      </div>
    </div>
  </div>
</Prerequisite>
`;

exports[`shared/components/Prerequisite Prerequisite component should work fine if need override and has checked override and no authority and has no error 1`] = `
<Prerequisite
  clearPrerequisiteErrsAction={[Function]}
  display={true}
  prerequisite={
    Immutable.Map {
      errors: Immutable.List [
      ],
      needOverride: true,
      userName: "",
      userPassword: "",
      isOverride: true,
      haveOverrideAuthority: false,
      overrideMessage: "override message",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateIsOverrideAction={[Function]}
  updateUserNameAction={[Function]}
  updateUserPasswordAction={[Function]}
>
  <div>
    <div
      className="prerequisite"
    >
      <Alert
        className="prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
        inverse={false}
        noClose={true}
        type="warning"
      >
        <div
          className="alert alert-warning prerequisite-warning aaui-flexbox u-flex-nowrap u-align-items-start"
          role="alert"
        >
          <span
            aria-label="warning icon"
            className="icon icon-exclamation-circle"
          />
          <div
            className="prerequisite-warning-content"
          >
            override message
            <Checkbox
              checked={true}
              className="prerequisite-warning-override"
              onChange={[Function]}
            >
              <CheckboxComponent
                checked={true}
                className="prerequisite-warning-override"
                onChange={[Function]}
              >
                <label
                  className="checkbox-wrapper prerequisite-warning-override"
                  htmlFor="checkbox_attr_for_3"
                >
                  <span
                    className="checkbox checkbox--checked"
                  >
                    <input
                      aria-checked={true}
                      checked={true}
                      className="checkbox__input"
                      id="checkbox_attr_for_3"
                      onChange={[Function]}
                      role="checkbox"
                      type="checkbox"
                    />
                    <span
                      className="checkbox__inner"
                    />
                    <span
                      className="checkbox__text"
                    >
                      Override
                    </span>
                  </span>
                </label>
              </CheckboxComponent>
            </Checkbox>
          </div>
        </div>
      </Alert>
      <div
        className="override-field"
      >
        <div
          className="override-field__user-name"
        >
          <span
            className="override-field__label "
          >
            Override User
          </span>
          <Input
            autoComplete="off"
            className="override-field__value"
            defaultValue=""
            errored={0}
            icon={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            size="md"
            type="text"
          >
            <div
              className="input-group input-group--md override-field__value"
            >
              <input
                autoComplete="off"
                className="input input-group__field input__field"
                defaultValue=""
                onBlur={[Function]}
                onChange={[Function]}
                onInput={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                type="text"
              />
            </div>
          </Input>
        </div>
        <div>
          <span
            className="override-field__label "
          >
            Password
          </span>
          <Input
            autoComplete="off"
            className="override-field__value"
            defaultValue=""
            errored={0}
            icon={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            size="md"
            type="password"
          >
            <div
              className="input-group input-group--md override-field__value"
            >
              <input
                autoComplete="off"
                className="input input-group__field input__field"
                defaultValue=""
                onBlur={[Function]}
                onChange={[Function]}
                onInput={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                type="password"
              />
            </div>
          </Input>
        </div>
      </div>
    </div>
  </div>
</Prerequisite>
`;
